src/main/java/hudson/plugins/sonar/utils/SonarUtils.java

```diff
@@ -66,12 +75,47 @@ private SonarUtils() {
    * Read logs of the build to find URL of the project dashboard in Sonar
    */
   public static String extractSonarProjectURLFromLogs(Run<?, ?> build) throws IOException {
+    return extractPatternFromLogs(URL_PATTERN_IN_LOGS, build);
+  }
+
+  public static <T extends Action> List<T> getPersistentActions(Actionable actionable, Class<T> type) {
+    List<T> filtered = new LinkedList<T>();
+
+    // we use this method to avoid recursively calling transitive action factories
+    for (Action a : actionable.getActions()) {
+      if (type.isAssignableFrom(a.getClass())) {
+        filtered.add((T) a);
+      }
+    }
+    return filtered;
+  }
+
+  public static Properties extractReportTask(Run<?, ?> build) throws IOException {
```

Comment:
I tend to prefer Map<String, String> over the old Properties object... but really minor (except maybe for Jenkins serialization in build action).